@xen-orchestra/web-core/lib/components/DonutChart.vue

```diff
@@ -0,0 +1,102 @@
+<!-- v1.1 -->
+<template>
+  <div class="wrapper">
+    <svg class="donut-chart" viewBox="0 0 100 100">
+      <circle class="segment max-value-segment" cx="50" cy="50" r="40" />
+      <circle
+        v-for="(segment, index) in computedSegments"
+        :key="index"
+        :class="segment.color"
+        :stroke-dasharray="`${segment.percent} ${circumference - segment.percent}`"
+        :stroke-dashoffset="segment.offset"
+        class="segment"
+        cx="50"
+        cy="50"
+        r="40"
+      />
+      <UiIcon :icon height="24" width="24" x="38" y="38" />
+    </svg>
+  </div>
+</template>
+
+<script setup lang="ts">
+import UiIcon from '@core/components/icon/UiIcon.vue'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+type ChartFigure = {
+  value: number
+  color: 'success' | 'warning' | 'error' | 'unknown'
+}
+
+type Segment = {
+  color: 'success' | 'warning' | 'error' | 'unknown'
+  percent: number
+  offset: number
+}
+
+const props = defineProps<{
+  segments: ChartFigure[]
+  maxValue?: number
+  icon?: IconDefinition
+}>()
```

Comment:
Let's keep consistency between the props and their type names.

```suggestion
type Segment = {
  value: number
  color: 'success' | 'warning' | 'error' | 'unknown'
}

type ComputedSegment = {
  color: 'success' | 'warning' | 'error' | 'unknown'
  percent: number
  offset: number
}

const props = defineProps<{
  segments: Segment[]
  maxValue?: number
  icon?: IconDefinition
}>()
```